src/diagnostics.rs

```diff
@@ -243,11 +253,16 @@ pub fn report_error<'tcx>(
                 labels.push(format!("this thread got stuck here"));
                 None
             }
-            GenmcStuckExecution => {
-                // This case should only happen in GenMC mode. We treat it like a normal program exit.
-                assert!(ecx.machine.data_race.as_genmc_ref().is_some());
-                tracing::info!("GenMC: found stuck execution");
-                return Some((0, true));
+            GenmcFinishedExecution => {
+                // This case should only happen in GenMC mode.
+                let genmc_ctx = ecx.machine.data_race.as_genmc_ref().unwrap();
+                // Check whether the execution got blocked or the program exited.
+                if let Some(exit_status) = genmc_ctx.get_exit_status() {
+                    // We have an exit status (from `exit(x)` or main thread return).
+                    return Some(exit_status);
```

Comment:
I'm not sure what happens here, but this sounds like information that should be attached to the error, not stored in global state.

I thought with `exit` we have to keep running other threads so throwing the exception is anyway not the right thing?